### PR TITLE
[receiver/datadog] Implement '/stats' endpoint

### PR DESCRIPTION
**Pull request by lopes-felipe**
**Description:** <Describe what has changed.>
Following up on #34772, this PR implements Datadog Agent's `/stats` endpoint, which is used for sending [client stats](https://github.com/DataDog/dd-trace-java/blob/master/communication/src/main/java/datadog/communication/ddagent/DDAgentFeaturesDiscovery.java#L40) of the [local sampled spans](https://github.com/DataDog/dd-trace-java/blob/master/dd-trace-core/src/main/java/datadog/trace/core/CoreTracer.java#L993-L1008) to calculate the [trace metrics](https://docs.datadoghq.com/tracing/metrics/metrics_namespace/).

This PR also does a minor refactor in the usage of Datadog's common headers, moving from literal strings to centralized constants.

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>

### CODE REVIEW

**Comment by lopes-felipe** [comment]
@jpkrohling, it looks like I'm having the same issues here. Have tried your `make lint` and `make gotidy` spells, but no luck. Any other magic words here that could please the CI Lords?

**Comment by jpkrohling** [comment]
I can run those for you, but can you try running `make gotidy` at the root of the project? It should take a while, but will run it on modules that consume this receiver, like the `otelcontribcol`. If you are using go 1.23, make sure the `toolchain` entry doesn't change (I typically need to revert  it manually).

File: cmd/otelcontribcol/go.mod [hunk 1]
```diff
@@ -4,7 +4,7 @@ module github.com/open-telemetry/opentelemetry-collector-contrib/cmd/otelcontrib
 
 go 1.22.0
 
-toolchain go1.22.6
+toolchain go1.22.4
```
Comment (jpkrohling) [inline cmd/otelcontribcol/go.mod:7]: this is likely what's causing `make check` to fail in the CI, while not failing locally -- try manually reverting this

**Review by jpkrohling — APPROVED** [review]

**Comment by jpkrohling** [comment]
Sorry about the merge conflict. Would you be able to address it? Once it's done, I'll merge on green.

**Pull request merged**